addons/account/tests/test_account_account.py

```diff
@@ -61,7 +61,8 @@ def test_toggle_reconcile(self):
 
         # Set the account as reconcile and fully reconcile something.
         account.reconcile = True
-        self.env['account.move.line'].invalidate_cache()
+        account.flush()
+        account.invalidate_cache()
```

Comment:
Should `flush` be called before every `invalidate_cache`? If so, could `invalidate_cache` simply call `flush` internally? If not, in what scenarios can `invalidate_cache` be called with/without `flush` be called?

Reply:
Also even if `flush` doesn't *always* need to be called before `invalidate_cache` (TBD though there are a pair of cases in this PR) maybe flushing should be done by default and opt-out?

aka
```python
    def invalidate_cache(self, fnames=None, ids=None, *, flush=True):
        if flush:
            self.flush()
        ...
```